x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/AddTests.java

```diff
@@ -303,9 +338,21 @@ private static Object addDatesAndTemporalAmount(Object lhs, Object rhs) {
             date = (Long) rhs;
             period = (TemporalAmount) lhs;
         }
+        return adder.applyAsLong(date, period);
+    }
+
+    private static long addMillis(Long date, TemporalAmount period) {
         return asMillis(asDateTime(date).plus(period));
     }
 
+    private static long addNanos(Long date, TemporalAmount period) {
+        return DateUtils.toLong(
+            Instant.from(
+                ZonedDateTime.ofInstant(DateUtils.toInstant(date), org.elasticsearch.xpack.esql.core.util.DateUtils.UTC).plus(period)
```

Comment:
I'm not thrilled that this is essentially copied from the production code. It means these tests are not doing a whole lot to ensure we are getting correct values out of the computation.  Unfortunately, I don't know of a more obviously correct way to check these results.  I think it's okay for a couple of reasons:

- A lot of these tests are meant to show that the function fits into the ecosystem correctly (e.g. has a good toString value, folds correctly, handles exactly the types it claims to handle, etc).
- We have a lot of CSV tests validating the actual date math
- The date math itself is being done by the java time library, which is quite well tested.

That said, I'm open to suggestions if anyone has a better idea.  Although clearly we made the same choice for millisecond date math, so I'm assuming we didn't have a better idea then. (comment applies to the verification function for the subtraction tests as well)